### PR TITLE
aws-azure-login: init at 3.3.0

### PR DESCRIPTION
**Pull request by yurrriq**
###### Motivation for this change

Add [aws-azure-login](https://github.com/sportradar/aws-azure-login).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/node-packages/node-env.nix [hunk 1]
```diff
@@ -444,11 +443,6 @@ let
         # Run post install hook, if provided
         runHook postInstall
       '';
-
```
Comment (yurrriq) [inline pkgs/development/node-packages/node-env.nix:447]: Is this deletion intentional?

File: pkgs/development/node-packages/composition.nix [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 let
   nodeEnv = import ./node-env.nix {
-    inherit (pkgs) stdenv python2 util-linux runCommand writeTextFile;
+    inherit (pkgs) stdenv python2 utillinux runCommand writeTextFile;
```
Comment (yurrriq) [inline pkgs/development/node-packages/composition.nix:9]: Not sure about this change...

**Comment by SuperSandro2000** [comment]
Please wait until we updated node2nix to 1.9.0 in nixpkgs.

**Comment by yurrriq** [comment]
> Please wait until we updated node2nix to 1.9.0 in nixpkgs.

Ref: #110813

File: pkgs/development/node-packages/node-env.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/node-packages/node-env.nix:447]: The generation script does not create it. It shouldn't be commited.

File: pkgs/development/node-packages/composition.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/node-packages/composition.nix:9]: This change breaks eval. If you update to the latest master you should get node2nix 1.9.0 which fixes this.

**Comment by yurrriq** [comment]
Result of `nixpkgs-review pr 110634` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
<br>- netlify-cli
</details>
<details>
  <summary>29 packages built:</summary>
<br>- antora
<br>- castnow
<br>- create-cycle-app
<br>- emojione
<br>- epgstation
<br>- etcher
<br>- iosevka
<br>- joplin
<br>- lumo
<br>- mastodon-bot
<br>- mirakurun
<br>- parity-ui
<br>- postcss-cli
<br>- redoc-cli
<br>- slack (slack-dark)
<br>- teleprompter
<br>- thelounge
<br>- vimPlugins.coc-diagnostic
<br>- vimPlugins.coc-metals
<br>- vimPlugins.coc-prettier
<br>- vimPlugins.coc-pyright
<br>- vimPlugins.coc-rust-analyzer
<br>- vimPlugins.coc-stylelint
<br>- vimPlugins.coc-tslint
<br>- vimPlugins.coc-vetur
<br>- vscode-extensions.matklad.rust-analyzer
<br>- vscode-extensions.ms-python.vscode-pylance
<br>- vscode-extensions.vadimcn.vscode-lldb
<br>- wasm-text-gen
</details>

File: pkgs/development/node-packages/default.nix [hunk 3]
```diff
@@ -13,6 +13,21 @@ let
         export NG_CLI_ANALYTICS=false
       '';
     };
+
+    aws-azure-login = super.aws-azure-login.override (
+      if stdenv.isDarwin
+      then {}
+      else {
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/default.nix:20]:
```suggestion
     {
```

this is not valid fox nixpkgs. Please use broken or platform instead.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/node-packages/default.nix (hunk 3 above)
Comment (yurrriq) [reply pkgs/development/node-packages/default.nix:20]: Thanks. I will open a separate PR to fix the other instances in this file then too.

**Comment by yurrriq** [comment]
Result of `nixpkgs-review pr 110634` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>netlify-cli</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>epgstation</li>
    <li>joplin</li>
    <li>redoc-cli</li>
  </ul>
</details>

**Comment by yurrriq** [comment]
was netlify-cli already broken, by chance? This seems more trouble than it's worth at this point.

**Comment by SuperSandro2000** [comment]
> was netlify-cli already broken, by chance? This seems more trouble than it's worth at this point.

yeah, since a bit longer.

**Comment by yurrriq** [comment]
> > was netlify-cli already broken, by chance? This seems more trouble than it's worth at this point.
> 
> yeah, since a bit longer.

Great, thanks! I think this is ready for review then. It would be nice if someone could test on Darwin. For now, I've just done `meta.platforms = pkgs.lib.platforms.linux`.

File: pkgs/development/node-packages/default.nix [hunk 4]
```diff
@@ -13,6 +13,19 @@ let
         export NG_CLI_ANALYTICS=false
       '';
     };
+
+    aws-azure-login = super.aws-azure-login.override {
+      meta.platforms = pkgs.lib.platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/default.nix:18]: Why is there no libs in inputs? That should be changed in the future.

File: pkgs/development/node-packages/default.nix [hunk 5]
```diff
@@ -13,6 +13,19 @@ let
...
+      nativeBuildInputs = [ pkgs.makeWrapper ];
+      prePatch = ''
+        export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1
+      '';
+      postInstall = ''
+        wrapProgram $out/bin/aws-azure-login \
+            --set PUPPETEER_EXECUTABLE_PATH ${pkgs.chromium.outPath}/bin/chromium
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/default.nix:25]:
```suggestion
            --set PUPPETEER_EXECUTABLE_PATH ${pkgs.chromium}/bin/chromium
```

It should auto find where the chromium binary is.

**Pull request merged**